resolwe/test/testcases/__init__.py

```diff
@@ -99,66 +55,78 @@ def _pre_setup(self, *args, **kwargs):
         ContentType.objects.clear_cache()
         super(TestCaseHelpers, self)._pre_setup(*args, **kwargs)
 
-    def _override_executor_settings(self):
-        """Override Resolwe Flow Executor's settings to avoid clashes.
-
-        When Resolwe's tests are run in parallel or from different
-        Resolwe's code bases on the same system (e.g. on a CI server),
-        we must ensure executor's test directories and containers have
-        unique names to avoid clashes.
-
-        """
-        flow_executor_settings = copy.copy(getattr(settings, 'FLOW_EXECUTOR', {}))
-
-        # Create unique executor's test directories.
-        executor_dirs = ('DATA_DIR', 'UPLOAD_DIR')
-        for exec_dir, new_dir_path in zip(
-                executor_dirs,
-                _test_dirs(*(flow_executor_settings[d] for d in executor_dirs))):
-            flow_executor_settings[exec_dir] = new_dir_path
-
-        # Update FLOW_DOCKER_MAPPINGS setting.
-        flow_docker_mappings = _get_updated_docker_mappings(flow_executor_settings)
-
-        # Create unique container name prefix.
-        flow_executor_settings['CONTAINER_NAME_PREFIX'] = '{}_{}_{}'.format(
-            flow_executor_settings.get('CONTAINER_NAME_PREFIX', 'resolwe'),
-            get_random_string(length=6),
-            os.path.basename(flow_executor_settings['DATA_DIR'])
-        )
-
-        self.settings = override_settings(FLOW_EXECUTOR=flow_executor_settings,
-                                          FLOW_DOCKER_MAPPINGS=flow_docker_mappings)
-
     def _clean_up(self):
         """Clean up after test."""
         if not self._keep_data:
+            # Do delete this here. See comment below near the makedirs
+            # in setUp.
             shutil.rmtree(settings.FLOW_EXECUTOR['DATA_DIR'], ignore_errors=True)
             shutil.rmtree(settings.FLOW_EXECUTOR['UPLOAD_DIR'], ignore_errors=True)
+            shutil.rmtree(settings.FLOW_EXECUTOR['RUNTIME_DIR'], ignore_errors=True)
 
         self.settings.disable()
 
     def setUp(self):
         """Prepare environment for test."""
         super(TestCaseHelpers, self).setUp()
 
-        self._override_executor_settings()
+        # Override flow executor settings
+        flow_executor_settings = copy.copy(getattr(settings, 'FLOW_EXECUTOR', {}))
+
+        # Get data directory settings
+        data_dir, upload_dir, runtime_dir = (
```

Comment:
If there's no need to update the Docker mappings (see below), then there's no need to define `{data,upload,runtime}_dir` variables.

Reply:
But there is a need to make sure the relevant directories exist, so there is a need for these variables regardless of what happens with docker mappings. Additionally, this code is functionally equivalent to the pre-refactor state, including the former reduncancy.